### PR TITLE
pyright but only for extensions

### PR DESCRIPTION
**Pull request by dni**
- add pyright workflow, unify mypy+flake8+pylint workflows
- fix pyright in Makefile, add pyright+pylint+flake8 to check target
- fix pyright lnbits/extensions
- exclude core from pyright

### CODE REVIEW

**Comment by prusnak** [comment]
Great idea!

File: lnbits/extensions/bleskomat/exchange_rates.py [hunk 1]
```diff
@@ -80,7 +80,5 @@ async def fetch_fiat_exchange_rate(currency: str, provider: str):
     else:
         data = {}
     getter = exchange_rate_providers[provider]["getter"]
-    print(getter)
-    if callable(getter):
-        rate = float(getter(data, replacements))
-    return rate
+    assert callable(getter), "cannot call getter function"
```
Comment (motorina0) [inline lnbits/extensions/bleskomat/exchange_rates.py:83]: - this replaces a `return None` with an `AssertionFailure`
- should be tested if it behaves correctly

File: lnbits/extensions/cashu/models.py [hunk 2]
```diff
@@ -68,10 +68,11 @@ def from_row(cls, row: Row):
     def from_dict(cls, d: dict):
         assert "secret" in d, "no secret in proof"
         assert "amount" in d, "no amount in proof"
+        assert "C" in d, "no C in proof"
         return cls(
-            amount=d.get("amount"),
-            C=d.get("C"),
-            secret=d.get("secret"),
+            amount=d["amount"],
```
Comment (motorina0) [inline lnbits/extensions/cashu/models.py:73]: Q: why is `[]` preferred to `.get("")`

Comment (dni) [reply lnbits/extensions/cashu/models.py:73]: get always returns an Optional and because they are required and asserted above its fine to directly access them.

File: lnbits/extensions/gerty/number_prefixer.py [hunk 3]
```diff
@@ -22,36 +23,29 @@ def si_classifier(val):
         -24: {"long_suffix": "yocto", "short_suffix": "y", "scalar": 10**-24},
     }
     exponent = int(math.floor(math.log10(abs(val)) / 3.0) * 3)
-    return suffixes.get(exponent, None)
+    suffix = suffixes.get(exponent)
+    assert suffix, f"could not classify: {val}"
+    return suffix
```
Comment (motorina0) [inline lnbits/extensions/gerty/number_prefixer.py:28]: > **Warning** now it throws instead of returning `None`

Comment (dni) [reply lnbits/extensions/gerty/number_prefixer.py:28]: i think that is actually accurate? @blackcoffeexbt

File: lnbits/extensions/lnaddress/views_api.py [hunk 4]
```diff
@@ -71,9 +71,7 @@ async def api_domain_create(
         if not cf_response or not cf_response["success"]:
             await delete_domain(domain.id)
             raise HTTPException(
-                status_code=HTTPStatus.BAD_REQUEST,
-                detail="Problem with cloudflare: "
-                + cf_response["errors"][0]["message"],
+                status_code=HTTPStatus.BAD_REQUEST, detail="Problem with cloudflare."
```
Comment (motorina0) [inline lnbits/extensions/lnaddress/views_api.py:74]: - nitpick: do not lose the response error
```py
logger.log("Cloudflare failed with: " + cf_response["errors"][0]["message"])
```

File: lnbits/extensions/streamalerts/crud.py [hunk 5]
```diff
@@ -208,8 +212,11 @@ async def service_add_token(service_id, token):
     is not overwritten.
     Tokens for Streamlabs never need to be refreshed.
     """
-    if (await get_service(service_id)).authenticated:
+    service = await get_service(service_id)
+    assert service
```
Comment (motorina0) [inline lnbits/extensions/streamalerts/crud.py:216]: - add error message to `assert`

Comment (dni) [reply lnbits/extensions/streamalerts/crud.py]: f547b1f1

File: lnbits/extensions/subdomains/cloudflare.py [hunk 6]
```diff
@@ -20,25 +18,21 @@ async def cloudflare_create_subdomain(
         "Content-Type": "application/json",
     }
     aRecord = subdomain + "." + domain.domain
```
Comment (motorina0) [inline lnbits/extensions/subdomains/cloudflare.py:20]: - the call in [tasks.py](https://github.com/lnbits/lnbits/blob/8bf410352331e46bf99a7cebc002159db668756e/lnbits/extensions/subdomains/tasks.py#L32) does not hand raised exceptions from `cloudflare_create_subdomain`

File: lnbits/extensions/subdomains/views_api.py [hunk 7]
```diff
@@ -121,21 +121,21 @@ async def api_subdomain_make_subdomain(domain_id, data: CreateSubdomain):
...
+            subdomain=data.subdomain,
+            record_type=data.record_type,
+            ip=data.ip,
+        )
         await cloudflare_deletesubdomain(
-            domain=domain, domain_id=cf_response["result"]["id"]
+            domain=domain, domain_id=res_json["result"]["id"]
         )
-    else:
+    except:
```
Comment (motorina0) [inline lnbits/extensions/subdomains/views_api.py:135]: - log the exception
```py
except Exception as ex:
   logger.warning(ex)
```

File: pyproject.toml [hunk 8]
```diff
@@ -94,6 +94,9 @@ include = [
 ]
 exclude = [
   "lnbits/wallets/lnd_grpc_files",
+  "lnbits/wallets",
```
[the remark this answers is not in the document]
Comment (dni) [reply pyproject.toml:97]: yes, i do just extensions for this PR all core later. so we can fast track extensions and the PR is smaller.

File: lnbits/extensions/tipjar/views_api.py [hunk 9]
```diff
@@ -55,25 +54,32 @@ async def api_create_tip(data: createTips):
...
+
     # Ensure that description string can be split reliably
     name = name.replace('"', "''")
     if not name:
         name = "Anonymous"
+
     description = f"{name}: {message}"
     charge = await create_charge(
-        user=charge_details["user"],
+        user=wallet.user,
```
Comment (motorina0) [inline lnbits/extensions/tipjar/views_api.py:73]: a bit hard to follow, but tests passed

**Review by motorina0 — APPROVED** [review]
LGTM, see comments

File: lnbits/extensions/tipjar/views_api.py (hunk 9 above)
Comment (dni) [reply lnbits/extensions/tipjar/views_api.py:73]: i removed the helper, which was bloated and only was used in 1 place, that simplified the whole part of the code a lot.

File: lnbits/extensions/bleskomat/exchange_rates.py (hunk 1 above)
Comment (dni) [reply lnbits/extensions/bleskomat/exchange_rates.py]: e0d9f2d7

File: lnbits/extensions/subdomains/views_api.py (hunk 7 above)
Comment (dni) [reply lnbits/extensions/subdomains/views_api.py]: e0d9f2d7

File: lnbits/extensions/subdomains/cloudflare.py (hunk 6 above)
Comment (dni) [reply lnbits/extensions/subdomains/cloudflare.py:20]: added try / except in tasks.py

File: lnbits/extensions/lnaddress/views_api.py (hunk 4 above)
Comment (dni) [reply lnbits/extensions/lnaddress/views_api.py:76]: 538f44e6

File: lnbits/extensions/gerty/number_prefixer.py (hunk 3 above)
Comment (dni) [reply lnbits/extensions/gerty/number_prefixer.py:28]: @blackcoffeexbt tested it
